test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/RunnableProjectGeneratorTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.Mount;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros.Config;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests.TemplateConfigTests;
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit;
+using static Microsoft.TemplateEngine.Orchestrator.RunnableProjects.RunnableProjectGenerator;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests
+{
+    public class RunnableProjectGeneratorTests : IClassFixture<EnvironmentSettingsHelper>
+    {
+        private EnvironmentSettingsHelper _environmentSettingsHelper;
+
+        public RunnableProjectGeneratorTests(EnvironmentSettingsHelper environmentSettingsHelper)
+        {
+            _environmentSettingsHelper = environmentSettingsHelper;
+        }
+
+        [Fact]
+        public async void CreateAsyncTest_GuidsMacroProcessingCaseSensitivity()
+        {
+            //
+            // Template content preparation
+            //
+
+            Guid inputTestGuid = new Guid("12aa8f4e-a4aa-4ac1-927c-94cb99485ef1");
+            string contentFileNamePrefix = "content - ";
+
+            string guidsConfigFormat = @"
+{{
+  ""guids"": [
+    ""{0}"" 
+  ]
+}}
+";
+            string guidsConfig = string.Format(guidsConfigFormat, inputTestGuid);
+
+            IDictionary<string, string?> templateSourceFiles = new Dictionary<string, string?>();
+            // template.json
+            templateSourceFiles.Add(TemplateConfigTestHelpers.DefaultConfigRelativePath, guidsConfig);
```

Comment:
just out of curiosity - is it enough to run the config? do we need to add name, sourceName etc?

Reply:
We are loosing some of the context info in the generator by using this trimmed config, however for a case of running the guid replacements it's enough.
Different test scenarios would need to add diferent config.